src/test/regress/sql/generated_identity.sql

```diff
@@ -0,0 +1,266 @@
+    c smallint GENERATED BY DEFAULT AS IDENTITY,
+    d serial,
+    e bigserial,
+    f smallserial,
+    g int
+);
+
+SELECT create_reference_table('generated_identities_test');
+
+\d generated_identities_test
+
+\c - - - :worker_1_port
+
+\d generated_identities.generated_identities_test
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO generated_identities_test (g)
+SELECT s FROM generate_series(11,20) s;
+
+SELECT * FROM generated_identities_test ORDER BY g;
+
+SELECT undistribute_table('generated_identities_test');
+
+\d generated_identities_test
+
+\c - - - :worker_1_port
+
+\d generated_identities.generated_identities_test
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+-- alter table .. add column .. GENERATED .. AS IDENTITY
+DROP TABLE IF EXISTS color;
+CREATE TABLE color (
+    color_name VARCHAR NOT NULL
+);
+SELECT create_distributed_table('color', 'color_name');
+ALTER TABLE color ADD COLUMN color_id BIGINT GENERATED ALWAYS AS IDENTITY;
+INSERT INTO color(color_name) VALUES ('Red');
+ALTER TABLE color ADD COLUMN color_id_1 BIGINT GENERATED ALWAYS AS IDENTITY;
+DROP TABLE color;
+
+-- insert data from workers
+CREATE TABLE color (
+    color_id BIGINT GENERATED ALWAYS AS IDENTITY UNIQUE,
+    color_name VARCHAR NOT NULL
+);
+SELECT create_distributed_table('color', 'color_id');
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+SELECT undistribute_table('color');
+SELECT create_distributed_table('color', 'color_id');
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+SELECT count(*) from color;
+
+-- modify sequence & alter table
+DROP TABLE color;
+
+CREATE TABLE color (
+    color_id BIGINT GENERATED ALWAYS AS IDENTITY UNIQUE,
+    color_name VARCHAR NOT NULL
+);
+SELECT create_distributed_table('color', 'color_id');
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+INSERT INTO color(color_name) VALUES ('Red');
+
+\c - - - :master_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+SELECT undistribute_table('color');
+
+ALTER SEQUENCE color_color_id_seq RENAME TO myseq;
+
+SELECT create_distributed_table('color', 'color_id');
+\ds+ myseq
+\ds+ color_color_id_seq
+\d color
+
+\c - - - :worker_1_port
+SET search_path TO generated_identities;
+SET client_min_messages to ERROR;
+
+\ds+ myseq
+\ds+ color_color_id_seq
```

Comment:
for future reference, it's not ideal to look psql command output because it has a tendency to change across versions. Some of the tests use test-specific views.

Reply:
@gokhangulbiz I had missed this point. Actually, I will insist more on this and say that we shouldn't do it like this for this PR. Let's avoid `\d` commands and query a `pg_catalog` table to print what we want, please.